acquisitions/src/main/resources/thunderjet/cross-modules/features/check-encumbrances-after-order-line-exchange-rate-update.feature

```diff
@@ -0,0 +1,237 @@
+
+  @Positive
+  Scenario: Check encumbrances after order line exchange rate update to manual rate 2 times
+    * def fundId = call uuid
+    * def budgetId = call uuid
+    * def orderId = call uuid
+    * def poLineId = call uuid
+    * def invoiceId = call uuid
+    * def invoiceLineId = call uuid
+
+    ### 1. Create finances
+    * configure headers = headersAdmin
+    * def v = call createFund { id: "#(fundId)" }
+    * def v = call createBudget { id: "#(budgetId)", allocated: 10000, fundId: "#(fundId)", status: "Active" }
+    * configure headers = headersUser
+
+    ### 2. Create an order and line
+    * def v = call createOrder { id: "#(orderId)" }
+    * def poLine = read('classpath:samples/mod-orders/orderLines/minimal-order-line.json')
+    * set poLine.id = poLineId
+    * set poLine.purchaseOrderId = orderId
+    * set poLine.fundDistribution[0].fundId = fundId
+    * set poLine.fundDistribution[0].code = fundId
+    * set poLine.cost.listUnitPrice = karate.get('listUnitPrice', 1.0)
+    * set poLine.cost.poLineEstimatedPrice = karate.get('listUnitPrice', 1.0)
+    * set poLine.isPackage = karate.get('isPackage', false)
+    * set poLine.titleOrPackage = karate.get('titleOrPackage', 'test')
+    * set poLine.paymentStatus = 'Pending'
+    Given path 'orders/order-lines'
+    And request poLine
+    When method POST
+    Then status 201
+
+    ### 3. Open the order
+    * def v = call openOrder { orderId: '#(orderId)' }
+
+   ###########################################################################################################
+
+    ### 4. Update the order line with first manual exchange rate
+    Given path 'orders/order-lines', poLineId
+    When method GET
+    Then status 200
+    * def poLine = $
+    * set poLine.cost.currency = 'AUD'
+    * set poLine.cost.exchangeRate = 0.7
+
+    Given path 'orders/order-lines', poLineId
+    And request poLine
+    When method PUT
+    Then status 204
+
+    ### 5. Check the budget after first manual exchange rate update
+    Given path '/finance/budgets'
+    And param query = 'fundId==' + fundId
+    When method GET
+    Then status 200
+    And match each $.budgets[*].available == 9999.3
+    And match each $.budgets[*].expenditures == 0
+    And match each $.budgets[*].encumbered ==  0.7
+    And match each $.budgets[*].awaitingPayment == 0
+    And match each $.budgets[*].unavailable == 0.7
+
+    ### 6. Check encumbrances after first manual exchange rate update
+    Given path '/finance/transactions'
+    And param query = 'encumbrance.sourcePurchaseOrderId == ' + orderId
+    When method GET
+    Then status 200
+    And match each $.transactions[*].currency == 'USD'
+    And match each $.transactions[*].encumbrance.orderStatus == 'Open'
+    And match each $.transactions[*].encumbrance.status == 'Unreleased'
+    And match each $.transactions[*].encumbrance.initialAmountEncumbered == 0.7
```

Comment:
we can also add assertion for encumbrance.amount. It should equals 0.7 in this case